src/daft-local-execution/src/udf_process_pool.rs

```diff
@@ -0,0 +1,369 @@
+pub(super) fn reset_process_pool_total_created() {
+    get_or_init_process_pool().reset_total_created();
+}
+
+/// Rust-side handle to a pool worker
+#[cfg(feature = "python")]
+pub(super) struct UdfWorkerHandle {
+    /// Python worker handle
+    py_handle: Py<PyAny>,
+    /// Set of active UDF names in this worker
+    active_udfs: HashSet<UDFName>,
+    /// Worker index for logging
+    index: usize,
+}
+
+#[cfg(feature = "python")]
+impl UdfWorkerHandle {
+    /// Create a new UDF worker handle
+    fn new(index: usize) -> DaftResult<Self> {
+        // Create Python worker handle
+        let py_handle = Python::attach(|py| {
+            Ok::<Py<PyAny>, PyErr>(
+                py.import(pyo3::intern!(py, "daft.execution.udf"))?
+                    .getattr(pyo3::intern!(py, "UdfWorkerHandle"))?
+                    .call0()?
+                    .unbind(),
+            )
+        })?;
+
+        let active_udfs = HashSet::new();
+
+        Ok(Self {
+            py_handle,
+            active_udfs,
+            index,
+        })
+    }
+
+    fn is_alive(&self) -> bool {
+        Python::attach(|py| -> bool {
+            self.py_handle
+                .bind(py)
+                .call_method0(pyo3::intern!(py, "is_alive"))
+                .and_then(|r| r.extract::<bool>())
+                .unwrap_or(false)
+        })
+    }
+
+    /// Submit a UDF task for execution
+    pub fn submit(
+        &mut self,
+        udf_name: &UDFName,
+        expr: &ExprRef,
+        input: RecordBatch,
+    ) -> DaftResult<(Series, OperatorMetrics)> {
+        use common_metrics::python::PyOperatorMetrics;
+        use daft_dsl::python::PyExpr;
+        use daft_recordbatch::python::PyRecordBatch;
+
+        // Check if this is a new UDF (needs initialization)
+        let is_new_udf = self.active_udfs.insert(udf_name.clone());
+
+        let result = Python::attach(|py| -> DaftResult<_> {
+            // Initialize the UDF if it's new
+            if is_new_udf {
+                let py_expr = PyExpr::from(expr.clone());
+                self.py_handle
+                    .bind(py)
+                    .call_method1(pyo3::intern!(py, "init_udf"), (udf_name.as_ref(), py_expr))?;
+            }
+
+            // Execute the UDF
+            let (py_result, py_stdout_lines, py_metrics) = self
+                .py_handle
+                .bind(py)
+                .call_method1(
+                    pyo3::intern!(py, "eval_input"),
+                    (udf_name.as_ref(), PyRecordBatch::from(input.clone())),
+                )?
+                .extract::<(PyRecordBatch, Vec<String>, PyOperatorMetrics)>()?;
+
+            Ok((
+                RecordBatch::from(py_result),
+                py_stdout_lines,
+                py_metrics.inner,
+            ))
+        });
+
+        // Print stdout lines if call succeeded
+        if let Ok((_, stdout_lines, _)) = &result {
+            let label = format!("[Pool Worker #{}]", self.index);
```

Comment:
I think before, this prefix would include the UDF that was currently running. Can we keep it? We have it from the `udf_name` parameter